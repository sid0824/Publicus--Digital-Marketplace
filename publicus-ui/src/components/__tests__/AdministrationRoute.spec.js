import toJson from 'enzyme-to-json';
import { AdministrationRoute } from "../AdministrationRoute";
import React from "react";
import {shallow, configure} from 'enzyme';
import { message } from 'antd';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('antd');

describe('Administration Route', () => {

  beforeEach(() => {

  });

  afterAll(() => {

  });

  test('should render authorised admin route', () => {
    const wrapper = shallow(
      <AdministrationRoute
        component={<div>Component</div>}
        authenticated
        adminUser
        other={'props'}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should redirect on unauthorised admin route', () => {
    const wrapper = shallow(
      <AdministrationRoute
        component={<div>Component</div>}
        authenticated={false}
        adminUser={false}
        other={'props'}
      />
    );
    const redirect = wrapper.props().render({ other: 'value '});
    expect(message.warning).toHaveBeenCalledWith('Login With an Admin Account To Access this Page');
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(shallow(redirect))).toMatchSnapshot();
  });

});