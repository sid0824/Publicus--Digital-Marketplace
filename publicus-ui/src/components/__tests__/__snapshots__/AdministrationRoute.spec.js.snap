// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Administration Route should redirect on unauthorised admin route 1`] = `
<Route
  other="props"
  render={[Function]}
/>
`;

exports[`Administration Route should redirect on unauthorised admin route 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Administration Route should render authorised admin route 1`] = `
<Route
  other="props"
  render={[Function]}
/>
`;
